plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRESTCatalog.java

```diff
@@ -0,0 +1,364 @@
+                                    id.name()))
+                            .collect(Collectors.toList()));
+        }
+        return tables.build();
+    }
+
+    @Override
+    public Transaction newCreateTableTransaction(ConnectorSession session, SchemaTableName schemaTableName, Schema schema, PartitionSpec partitionSpec, String location, Map<String, String> properties)
+    {
+        return sessionCatalog.buildTable(convert(session), toIdentifier(schemaTableName), schema)
+                .withPartitionSpec(partitionSpec)
+                .withLocation(location)
+                .withProperties(properties)
+                .createTransaction();
+    }
+
+    @Override
+    public void dropTable(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        if (!sessionCatalog.dropTable(convert(session), toIdentifier(schemaTableName))) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR, format("Failed to drop table: %s", schemaTableName));
+        }
+    }
+
+    @Override
+    public void renameTable(ConnectorSession session, SchemaTableName from, SchemaTableName to)
+    {
+        try {
+            sessionCatalog.renameTable(convert(session), toIdentifier(from), toIdentifier(to));
+        }
+        catch (RESTException e) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR, e);
+        }
+    }
+
+    @Override
+    public Table loadTable(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        try {
+            return tableCache.computeIfAbsent(
+                    schemaTableName.toString(),
+                    key -> {
+                        BaseTable baseTable = (BaseTable) sessionCatalog.loadTable(convert(session), toIdentifier(schemaTableName));
+                        return new BaseTable(baseTable.operations(), IcebergUtil.quotedTableName(schemaTableName));
+                    });
+        }
+        catch (NoSuchTableException e) {
+            throw new TableNotFoundException(schemaTableName, e);
+        }
+        catch (Exception e) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR, e);
+        }
+    }
+
+    @Override
+    public void updateTableComment(ConnectorSession session, SchemaTableName schemaTableName, Optional<String> comment)
+    {
+        Table icebergTable = sessionCatalog.loadTable(convert(session), toIdentifier(schemaTableName));
+        if (comment.isEmpty()) {
+            icebergTable.updateProperties().remove(TABLE_COMMENT).commit();
+        }
+        else {
+            icebergTable.updateProperties().set(TABLE_COMMENT, comment.get()).commit();
+        }
+    }
+
+    @Override
+    public String defaultTableLocation(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        return null;
+    }
+
+    @Override
+    public void setTablePrincipal(ConnectorSession session, SchemaTableName schemaTableName, TrinoPrincipal principal)
+    {
```

Comment:
Add `        throw new TrinoException(NOT_SUPPORTED, "setTablePrincipal is not supported for Iceberg REST catalogs");`